dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/ChatCompletions/AIAgentChatCompletionsProcessor.cs

```diff
@@ -0,0 +1,108 @@
+
+        var inputItems = chatCompletionOptions.GetMessages();
+        var chatMessages = inputItems.AsChatMessages();
+
+        if (chatCompletionOptions.GetStream())
+        {
+            return new OpenAIStreamingChatCompletionResult(this._agent, chatMessages);
+        }
+
+        var agentResponse = await this._agent.RunAsync(chatMessages, agentThread, options, cancellationToken).ConfigureAwait(false);
+        return new OpenAIChatCompletionResult(agentResponse);
+    }
+
+    private sealed class OpenAIChatCompletionResult(AgentRunResponse agentRunResponse) : IResult
+    {
+        public async Task ExecuteAsync(HttpContext httpContext)
+        {
+            // note: OpenAI SDK types provide their own serialization implementation
+            // so we cant simply return IResult wrap for the typed-object.
+            // instead writing to the response body can be done.
+
+            var cancellationToken = httpContext.RequestAborted;
+            var response = httpContext.Response;
+
+            var chatResponse = agentRunResponse.AsChatResponse();
+            var openAIChatCompletion = chatResponse.AsOpenAIChatCompletion();
+            var openAIChatCompletionJsonModel = openAIChatCompletion as IJsonModel<ChatCompletion>;
+            Debug.Assert(openAIChatCompletionJsonModel is not null);
+
+            var writer = new Utf8JsonWriter(response.BodyWriter, new JsonWriterOptions { SkipValidation = false });
+            openAIChatCompletionJsonModel.Write(writer, ModelReaderWriterOptions.Json);
+            await writer.FlushAsync(cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+#pragma warning disable CS9113 // Parameter is unread.
+    private sealed class OpenAIStreamingChatCompletionResult(AIAgent agent, IEnumerable<ChatMessage> chatMessages) : IResult
+#pragma warning restore CS9113 // Parameter is unread.
+    {
+        public Task ExecuteAsync(HttpContext httpContext)
+        {
+            var cancellationToken = httpContext.RequestAborted;
+            var response = httpContext.Response;
+
+            // Set SSE headers
+            response.Headers.ContentType = "text/event-stream";
+            response.Headers.CacheControl = "no-cache,no-store";
+            response.Headers.Connection = "keep-alive";
+            response.Headers.ContentEncoding = "identity";
+            httpContext.Features.GetRequiredFeature<IHttpResponseBodyFeature>().DisableBuffering();
+
+            return SseFormatter.WriteAsync(
+                source: this.GetStreamingResponsesAsync(cancellationToken),
+                destination: response.Body,
+                itemFormatter: (sseItem, bufferWriter) =>
+                {
+                    var sseDataJsonModel = (IJsonModel<StreamingChatCompletionUpdate>)sseItem.Data;
+                    var json = sseDataJsonModel.Write(ModelReaderWriterOptions.Json);
+                    bufferWriter.Write(json);
+                },
+                cancellationToken);
```

Comment:
IIRC, the chat completion SSE stream is supposed to end with a "[done]" message?

Reply:
Interesting. Ok. I really thought it was necessary for chat completions as well.